pkg/helper/object_meta_merger.go

```diff
@@ -1,18 +1,22 @@
 package helper
 
-import (
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-)
+import "github.com/3scale/3scale-operator/pkg/common"
 
 // From
 // https://github.com/openshift/library-go/blob/master/pkg/operator/resource/resourcemerge/object_merger.go
 
 // EnsureObjectMeta ensure Labels, Annotations
-func EnsureObjectMeta(existing, desired *metav1.ObjectMeta) bool {
+func EnsureObjectMeta(existing, desired common.KubernetesObject) bool {
 	updated := false
 
-	MergeMapStringString(&updated, &existing.Labels, desired.Labels)
-	MergeMapStringString(&updated, &existing.Annotations, desired.Annotations)
+	existingLabels := existing.GetLabels()
+	existingAnnotations := existing.GetAnnotations()
+
+	MergeMapStringString(&updated, &existingLabels, desired.GetLabels())
+	MergeMapStringString(&updated, &existingAnnotations, desired.GetAnnotations())
+
+	existing.SetLabels(existingLabels)
```

Comment:
I think the changes in the body of this method are not needed.

If you look at the MergeMapStringString, the initialization and set is performed (it's doing a merge) there.

See https://github.com/3scale/3scale-operator/pull/356#discussion_r416040478 where this was commented

Reply:
Note that the function signature has been changed. The generic reconciler is based on `common.KubernetesObject` interface. That interface does not implement `metav1.ObjectMeta`. You no longer have direct access to labels and annotations. Instead, you have getters/setters methods for that.

Reply:
I noticed the signature change. I was not aware of the common.KubernetesObject interface not implementing metav1.ObjectMeta :+1: